tests/io/v3/assets/test_assets_api.py

```diff
@@ -0,0 +1,460 @@
+'''
+Testing the Assets endpoint
+'''
+import pytest
+import responses
+from requests import Response
+from responses import matchers
+
+from tenable.io.v3.base.iterators.explore_iterator import (CSVChunkIterator,
+                                                           SearchIterator)
+
+ASSET_BASE_URL = r'https://cloud.tenable.com/api/v3/assets'
+BASE_URL = 'https://cloud.tenable.com/api/v3'
+
+
+@responses.activate
+def test_search(api):
+    '''
+    Test the search method
+    '''
+    fields = ['netbios_name', 'acr_score', 'id']
+    sort = [('last_seen', 'desc')]
+    filters = ('last_scan_target', 'eq', '192.0.2.57')
+
+    payload = {
+        'fields': fields,
+        'limit': 200,
+        'sort': [{'last_seen': 'desc'}],
+        'filter': {
+            'property': 'last_scan_target',
+            'operator': 'eq',
+            'value': '192.0.2.57',
+        },
+    }
+
+    api_response = {
+        'assets': [
+            {
+                'id': '116af8c3-969d-4621-9f9f-364eeb58e3a7',
+                'has_agent': False,
+                'last_seen': '2018-12-31T15:00:57.000Z',
+                'last_scan_target': '192.0.2.57',
+                'sources': ['source_test'],
+                'acr_score': 213,
+                'acr_drivers': [],
+                'exposure_score': 753,
+                'scan_frequency': [],
+                'ipv4': ['0.0.0.0'],
+                'ipv6': [],
+                'fqdn': ['example.test'],
+                'netbios_name': ['SCCM'],
+                'operating_system': ['DOS'],
+                'agent_name': ['test'],
+                'aws_ec2_name': ['test_aws'],
+                'mac_address': ['00:00:00:00:00:00'],
+            }
+        ],
+        'pagination': {'total': 1, 'next': 'nextToken'},
```

Comment:
doesn't make sense to set **next** token value, when total number of records is only 1.